DRAM access bug in test

**Describe the bug**
Hardcoded core coordinate intermittently fails on p100a (harvested DRAM).

`test_remote_tt_device` (nanobind/tests/test_py_tt_device.py) reads a hardcoded
coordinate via `noc_read32(9, 0, 0)`. On NOC0, `(9, 0)` is a DRAM endpoint (`d9-0`),
and on some SKUs that DRAM bank is harvested — confirmed on P100/p100a. `noc_read32`
treats its args as raw LITERAL NOC0 coordinates with no harvesting-aware translation,
so the read targets a dead/harvested endpoint, stalls (~203 ms for a 4-byte load),
and aborts with `DeviceTimeoutError` once the per-op MMIO budget elapses.

It is SKU/board dependent: it fails on p100a (where `d9-0` is harvested) and passes
on p150b (where it is live). The other tests in the same file avoid this by sourcing
a live core from the descriptor: `SocDescriptor.get_cores(CoreType.TENSIX, ...)[0]`,
which excludes harvested cores.

**Environment**
- UMD commit (branch if applicable): main 
- FW version: 19.8.0
- KMD version: 2.8.0
- Board/system type: P100 (CI runner `tt-ubuntu-2204-p100a-viommu-stable`, IOMMU enabled)

**Logs**
Harvesting mask on the affected p100a (from TestCluster.PrintAllSiliconChipsAllCores):

    IOMMU: enabled (cluster.cpp:142)
    KMD version: 2.8.0 (cluster.cpp:145)
    ...
    Harvested DRAM core d9-0 (NOC0)
    Harvested DRAM core d9-1 (NOC0)
    Harvested DRAM core d9-11 (NOC0)

Pytest failure:

    FAILED tests/test_py_tt_device.py::TestTTDevice::test_remote_tt_device
      - tt_umd.tt_umd.error.DeviceTimeoutError: MMIO per-op timeout: 4B load took
        203387 us (budget=10 ms), 4 of 4 bytes remaining.
    1 failed, 26 passed, 5 skipped

**Stack trace (optional)**

    tt_umd.tt_umd.error.DeviceTimeoutError: MMIO per-op timeout: 4B load took 203387 us (budget=10 ms), 4 of 4 bytes remaining.
    Location: device/pcie/device_memcpy.cpp:38
     1. tt::umd::memcpy_from_device(void*, void const volatile*, unsigned long, std::function<bool ()> const&)
     2. tt::umd::TlbWindow::read_block_reconfigure(void*, tt::xy_pair, unsigned long, unsigned long, tt::umd::NocId, unsigned long)
     3. tt::umd::PcieProtocol::read_from_device(void*, tt::xy_pair, unsigned long, unsigned long, tt::umd::NocId)
     4. tt::umd::TTDevice::read_from_device(void*, tt::umd::CoreCoord, unsigned long, unsigned long)
    (remaining frames are Python interpreter frames from the nanobind call into noc_read32)

**To reproduce**
1. On a P100 where DRAM `d9-0` is harvested (e.g. some of the p100a CI runner).
2. Build UMD and install the wheel.
3. Run the nanobind Python tests: `python -m pytest -s tests` (nanobind/tests).
4. `test_remote_tt_device` fails with `DeviceTimeoutError` reading core (9,0).

**Expected behavior**
The test should read a guaranteed-live core (e.g. `SocDescriptor.get_cores(CoreType.TENSIX, ...)[0]`,
which excludes harvested cores) so it passes on all SKUs regardless of harvesting.
Reading a hardcoded coordinate that can land on a harvested endpoint should not be
part of the test's happy path.

--- a/nanobind/tests/test_py_tt_device.py
+++ b/nanobind/tests/test_py_tt_device.py
@@ -293,8 +293,15 @@ class TestTTDevice(unittest.TestCase):
                     f"Remote device {chip} should be remote",
                 )
 
-            val = umd_tt_devices[chip].noc_read32(9, 0, 0)
-            print(f"Read value from device, core 9,0 addr 0x0: {val}")
+            tt_device = umd_tt_devices[chip]
+            soc_descriptor = tt_umd.SocDescriptor(tt_device)
+            tensix_core = soc_descriptor.get_cores(
+                tt_umd.CoreType.TENSIX, tt_umd.CoordSystem.TRANSLATED
+            )[0]
+            val = tt_device.noc_read32(tensix_core.x, tensix_core.y, 0)
+            print(
+                f"Read value from device {chip}, core {tensix_core.x},{tensix_core.y} addr 0x0: {val}"
+            )
 
     def test_read_kmd_version(self):
         # Test reading KMD version

